core/src/main/java/org/elasticsearch/plugins/PluginManager.java

```diff
@@ -430,10 +430,8 @@ public void listInstalledPlugins(Terminal terminal) throws IOException {
                 // Elasticsearch new download service uses groupId org.elasticsearch.plugins from 2.0.0
                 if (user == null) {
                     // TODO Update to https
-                    if (Version.CURRENT.snapshot()) {
-                        addUrl(urls, String.format(Locale.ROOT, "http://download.elastic.co/elasticsearch/snapshot/org/elasticsearch/plugin/%s/%s-SNAPSHOT/%s-%s-SNAPSHOT.zip", repo, version, repo, version));
-                    }
                     addUrl(urls, String.format(Locale.ROOT, "http://download.elastic.co/elasticsearch/release/org/elasticsearch/plugin/%s/%s/%s-%s.zip", repo, version, repo, version));
+                    addUrl(urls, String.format(Locale.ROOT, "http://download.elastic.co/elasticsearch/staging/org/elasticsearch/plugin/%s/%s/%s-%s.zip", repo, version, repo, version));
```

Comment:
Yes. A flag looks a good solution to me.
We could also autodetect SNAPSHOT in the provided String and add all SNAPSHOT repos in that case.

Can you add Sonatype SNAPSHOT repo BTW?